arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -1010,26 +1051,24 @@ size_t IResearchViewExecutor<ordered, materializeType>::skip(size_t limit) {
 
   size_t const toSkip = limit;
 
-  for (size_t count = this->_reader->size(); _readerOffset < count;) {
+  for (size_t count = this->_reader->size(); _readerOffset < count; ++_readerOffset) {
     if (!_itr && !resetIterator()) {
```

Comment:
There is a difference here compared to the previous version:
If the _if_ condition `(!_itr && !resetIterator())` is satisfied, we _now_ increase `_readerOffset` and try again.
Previously, if the _if_ condition was satisfied, we did *not* increase `_readerOffset`.
The change may be intentional, I am just pointing this out in case it is an unintended change.

Reply:
It is intentional. Initial version will loop forever if resetIterator fails and failing segment will never be skipped. That was a bug actually 😅

Reply:
Ok, that was what I expected. :+1: